src/main/java/address/controller/PersonOverviewController.java

```diff
@@ -67,9 +74,12 @@ public void setConnections(MainController mainController, ModelManager modelMana
     private void handleDeletePerson() {
         int selectedIndex = personList.getSelectionModel().getSelectedIndex();
         if (selectedIndex >= 0) {
+            Person tmpPerson = personList.getItems().get(selectedIndex);
             mainController.getStatusBarHeaderController().postStatus(
-                                                    new PersonDeletedStatus(personList.getItems().get(selectedIndex)));
-            modelManager.deletePerson(personList.getItems().get(selectedIndex));
+                                                    new PersonDeletedStatus(tmpPerson));
+            personList.getItems().get(selectedIndex).setIsDeleted(true);
+            requestExecutor.schedule(()
+                    -> Platform.runLater(() -> modelManager.deletePerson(tmpPerson)), 3, TimeUnit.SECONDS);
```

Comment:
Platform.runLater returns instantly and queues the task on the UI thread, shouldn't use an executor here